ResearchSuite/ResearchSuite/RSDTrackedSelectionDataSource.swift

```diff
@@ -0,0 +1,211 @@
+    ///     - taskPath:         The current task path for this data source.
+    public init(step: RSDTrackedItemsStep, taskPath: RSDTaskPath) {
+        self.step = step
+        self.taskPath = taskPath
+        
+        // Look for an initial result in either the taskPath or on self.
+        var initialResult: RSDTrackedItemsResult?
+        if let result = step.result {
+            initialResult = result
+        }
+        else if let previousResult = taskPath.previousResults?.rsd_last(where: { $0.identifier == step.identifier }) as? RSDTrackedItemsResult {
+            initialResult = previousResult
+        }
+        self.initialResult = initialResult
+        
+        // build the sections and groups
+        let (sections, groups) = type(of: self).buildSections(step: step, initialResult: initialResult)
+        self.sections = sections
+        self.itemGroups = groups
+    }
+    
+    open class func buildSections(step: RSDTrackedItemsStep, initialResult: RSDTrackedItemsResult?) -> ([RSDTableSection], [RSDChoicePickerTableItemGroup]) {
+
+        let sectionItems = step.sections ?? []
+        let dataType: RSDFormDataType = .collection(.multipleChoice, .string)
+        var trackedItems = step.items
+        var trackedAnswers = initialResult?.selectedAnswers ?? []
+        var tableSections: [RSDTableSection] = []
+        var itemGroups: [RSDChoicePickerTableItemGroup] = []
+        
+        func appendSection(choices: [RSDTrackedItem], section: RSDTrackedSection?) {
+            let identifier = section?.identifier ?? step.identifier
+            let idx = tableSections.count
+            let field = RSDChoiceInputFieldObject(identifier: identifier, choices: choices, dataType: dataType, uiHint: .list)
+            // group
+            let group = RSDChoicePickerTableItemGroup(beginningRowIndex: 0, inputField: field, uiHint: .list, choicePicker: field)
+            group.sectionIndex = idx
+            itemGroups.append(group)
+            // table section
+            let tableSection = RSDTableSection(sectionIndex: idx, tableItems: group.items)
+            tableSection.title = (section?.text ?? section?.identifier) ?? (
+                idx > 0 ? Localization.localizedString("OTHER_SECTION_TITLE") : nil
+            )
+            tableSection.subtitle = section?.detail
+            tableSections.append(tableSection)
+            // selection state
+            let choiceIdentifiers = choices.map { $0.identifier }
+            let answers = trackedAnswers.remove(where: { choiceIdentifiers.contains($0.identifier) }).map { $0.identifier }
+            let selectableItems = group.items as! [RSDChoiceTableItem]
+            for input in selectableItems {
+                input.selected = answers.contains((input.choice as! RSDTrackedItem).identifier)
+            }
+            try! group.setAnswer(answers)
+        }
+        
+        /// Look through the sections first for a mapped item
```

Comment:
Should these *not* be doc comments? (from here to the end of the buildSections func)